R/tm_a_regression.R

```diff
@@ -1000,7 +1076,7 @@ srv_a_regression <- function(id,
 
     teal.widgets::verbatim_popup_srv(
       id = "rcode",
-      verbatim_content = reactive(teal.code::get_code(output_q())),
+      verbatim_content = reactive(teal.code::get_code(req(output_q()))),
```

Comment:
If you open an app too fast, and you click show r code, in some circumnstances you can see a warning. That's why we have `req` in here. You can see the same by running below example (no `req` at `get_code`)

```r
interactive_decorator_lang <- teal_transform_module(
  ui = function(id) {
    ns <- NS(id)
    div(
      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
    )
  },
  server = make_teal_transform_server(
    expression(
      plot <- plot +
        ggtitle("This is title") +
        xlab(x_axis_title)
    )
  )
)

tm_decorated_plot <- function(label = "module", transformators = list(), decorators = teal_transform_module()) {
  module(
    label = label,
    ui = function(id, decorators) {
      ns <- NS(id)
      div(
        selectInput(ns("dataname"), label = "select dataname", choices = NULL),
        selectInput(ns("x"), label = "select x", choices = NULL),
        selectInput(ns("y"), label = "select y", choices = NULL),
        ui_teal_transform_data(ns("decorate"), transformators = decorators),
        plotOutput(ns("plot")),
        verbatimTextOutput(ns("text"))
      )
    },
    server = function(id, data, decorators) {
      moduleServer(id, function(input, output, session) {
        observeEvent(data(), {
          updateSelectInput(inputId = "dataname", choices = names(data()))
        })
        
        observeEvent(input$dataname, {
          req(input$dataname)
          updateSelectInput(inputId = "x", choices = colnames(data()[[input$dataname]]))
          updateSelectInput(inputId = "y", choices = colnames(data()[[input$dataname]]))
        })
        
        plot_data <- debounce(
          reactive({
            req(input$dataname, input$x, input$y)
            within(data(),
                   {
                     plot <- ggplot2::ggplot(dataname, ggplot2::aes(x = x, y = y)) +
                       ggplot2::geom_point()
                   },
                   dataname = as.name(input$dataname),
                   x = as.name(input$x),
                   y = as.name(input$y)
            )
          }), 200
        )
        
        plot_data_decorated <- srv_teal_transform_data("decorate", data = plot_data, transformators = decorators)
        
        plot_r <- reactive({
          req(plot_data_decorated())
          plot_data_decorated()[["plot"]]
        })
        
        output$plot <- renderPlot(plot_r())
        output$text <- renderText({
          teal.code::get_code(plot_data_decorated())
        })
      })
    },
    ui_args = list(decorators = decorators),
    server_args = list(decorators = decorators)
  )
}

library(ggplot2)
app <- init(
  data = teal_data(iris = iris, mtcars = mtcars),
  modules = modules(
    tm_decorated_plot("identity"),
    tm_decorated_plot("lang", decorators = static_decorator_lang)
  )
)

if (interactive()) {
  shinyApp(app$ui, app$server)
}
```